### PR TITLE
DOC: Improve doc strings

### PR DESCRIPTION
**Pull request by bashtage**
Improve types and correct errors

- [ ] closes #xxxx
- [ ] tests added / passed. 
- [ ] code/documentation is well formatted.  
- [ ] properly formatted commit message. See 
      [NumPy's guide](https://docs.scipy.org/doc/numpy-1.15.1/dev/gitwash/development_workflow.html#writing-the-commit-message). 

<details>

**Notes**:

* It is essential that you add a test when making code changes. Tests are not 
  needed for doc changes.
* When adding a new function, test values should usually be verified in another package (e.g., R/SAS/Stata).
* When fixing a bug, you must add a test that would produce the bug in master and
  then show that it is fixed with the new code.
* New code additions must be well formatted. Changes should pass flake8. If on Linux or OSX, you can
  verify you changes are well formatted by running 
  ```
  git diff upstream/master -u -- "*.py" | flake8 --diff --isolated
  ```
  assuming `flake8` is installed. This command is also available on Windows 
  using the Windows System for Linux once `flake8` is installed in the 
  local Linux environment. While passing this test is not required, it is good practice and it help 
  improve code quality in `statsmodels`.
* Docstring additions must render correctly, including escapes and LaTeX.

</details>

### CODE REVIEW

File: statsmodels/imputation/mice.py [hunk 1]
```diff
@@ -767,7 +767,7 @@ def plot_fit_obs(self, col_name, lowess_args=None,
             containing x-axis jitter and y-axis jitter, respectively.
         plot_points : bool
             If True, the data points are plotted.
-        ax : matplotlib axes object
+        ax : AxesSubplot
```
Comment (josef-pkt) [inline statsmodels/imputation/mice.py]: why do you remove `matplotlib` everywhere?
AFAIK we only support matplotlib

File: statsmodels/graphics/gofplots.py [hunk 2]
```diff
@@ -22,10 +22,16 @@ class ProbPlot(object):
     Parameters
     ----------
     data : array_like
-        1d data array
-    dist : A scipy.stats or statsmodels distribution
-        Compare x against dist. The default is
-        scipy.stats.distributions.norm (a standard normal).
+        A 1d data array
+    dist : callable
```
Comment (josef-pkt) [inline statsmodels/graphics/gofplots.py]: I think callable is not correct.
We need a distribution with appropriate methods but it doesn't have to be callable, AFAIR

File: statsmodels/imputation/mice.py (hunk 1 above)
Comment (bashtage) [reply statsmodels/imputation/mice.py]: It is not needed.  It also renders poorly.

File: statsmodels/graphics/gofplots.py (hunk 2 above)
Comment (bashtage) [reply statsmodels/graphics/gofplots.py]: All of the referenced distribution are callable. The remainder fo the docstring provides the explanation of what is required. It doesn't make sense to try and stick this into a type, which it isn't.

File: statsmodels/imputation/mice.py (hunk 1 above)
Comment (bashtage) [reply statsmodels/imputation/mice.py]: Types like Series, Figure, ndarray, DataFrame, Axes, AxesSubPlot are all distinctive enough to need to further qualification.

**Comment by bashtage** [comment]
Search for Checking Doc Strings in https://api.travis-ci.org/v3/job/629133667/log.txt

These are all not very good docstrings (some quite bad, like missing many arguments).

There are around 3000 issues.

**Comment by josef-pkt** [comment]
some of those are "silly" 
`Logit.hessian: Return value description should finish with "."`
or `Return value description should start with a capital letter`

I removed some points `.` in my docstrings to avoid the line length restriction if it was just 1 character
(79 instead of 80 characters)

My impression is that some of the changes make the description less precise and less specific just to satisfy some docstring pep-8.

But the `log.txt` list is good for "real" problems like missing pieces in docstrings.

**Pull request merged**